[17.0][ADD] caldav_sync: sync Odoo with CalDAV servers

mdurepos opened the pull request:
Allows synchronization of calendar events between Odoo and CalDAV servers, configurable per user.

Back-ported from 18.0. Originally available on
git.bemade.org/bemade/bemade-addons (and apps.odoo.com).

mdurepos committed:
[17.0][ADD] caldav_sync: sync Odoo with CalDAV servers

Back-ported from 18.0. Originally available on
git.bemade.org/bemade/bemade-addons (and apps.odoo.com).

XtremXpert reviewed (approved):
working before migarte to 18.0

mdurepos committed:
[17.0] caldav_sync: fix recurrence bug

Fixes a previously unnoticed bug where splitting a recurrenc chain
caused infinite recursion. Once that fix was made, another bug
surfaced where an empty RRULE string was passed to the CalDAV server
under the same conditions.
